crates/subspace-networking/src/utils.rs

```diff
@@ -141,7 +141,7 @@ impl SemState {
         if let Some(dec) = self.usage.checked_sub(1) {
             self.usage = dec;
         } else {
-            panic!("SemState::free_one(): invalid free, state = {self:?}");
+            warn!(state = ?self, "SemState::free_one(): invalid free");
```

Comment:
This has to stay a panic, as the permits are freed from the drop() (going out of scope, or user explicitly doing std::mem::drop() on the permit). We should never have a situation where we can free more permits than allocated, unlike the expand()/shrink() flow